pkg/clients/cloudsql/cloudsql.go

```diff
@@ -283,10 +296,17 @@ func LateInitializeSpec(spec *v1beta1.CloudSQLInstanceParameters, in sqladmin.Da
 
 // IsUpToDate checks whether current state is up-to-date compared to the given
 // set of parameters.
-func IsUpToDate(in *v1beta1.CloudSQLInstanceParameters, currentState sqladmin.DatabaseInstance) bool {
-	currentParams := &v1beta1.CloudSQLInstanceParameters{}
-	LateInitializeSpec(currentParams, currentState)
-	return cmp.Equal(in, currentParams, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
+func IsUpToDate(name string, in *v1beta1.CloudSQLInstanceParameters, currentState *sqladmin.DatabaseInstance) (bool, error) {
+	genState, err := copystructure.Copy(currentState)
+	if err != nil {
+		return false, err
+	}
+	_, ok := genState.(*sqladmin.DatabaseInstance)
```

Comment:
I think you can do something like `generatedState, ok := ..` and use `generatedState` directly in `GenerateDatabaseInstance` without needing to cast it again.

Reply:
I wish this were true, but unfortunately the returned value is not of the type that the type assertion checks for here.

Reply:
@muvaf Actually, you are completely correct. My statement above is false, I had been setting it to the same var as the one that was being casted, which made it continue to be treated as an interface. Thanks for pointing this out!